Upstream: yes
Reason: Fixes the tests with proj 9.1.1

From f4b2b195c2d1901d25c91fb0af018034f7932a71 Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sat, 26 Nov 2022 13:23:12 +0100
Subject: [PATCH 1/2] listgeo: add a -no_corners option to avoid printing
 corner coordinates which might be PROJ version sensitive (refs #81)

---
 libgeotiff/bin/listgeo.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libgeotiff/bin/listgeo.c b/libgeotiff/bin/listgeo.c
index 06c45f7..acad54c 100644
--- a/libgeotiff/bin/listgeo.c
+++ b/libgeotiff/bin/listgeo.c
@@ -29,6 +29,7 @@ void Usage()
         "  -tfw: Generate a .tfw (ESRI TIFF World) file for the target file.\n"
         "  -proj4: Report PROJ.4 equivalent projection definition.\n"
         "  -no_norm: Don't report 'normalized' parameter values.\n"
+        "  -no_corners: Don't report corner coordinates.\n"
         "  filename: Name of the GeoTIFF file to report on.\n" );
         
     exit( 1 );
@@ -42,6 +43,7 @@ int main(int argc, char *argv[])
     int		i, norm_print_flag = 1, proj4_print_flag = 0;
     int		tfw_flag = 0, inv_flag = 0, dec_flag = 0;
     int         st_test_flag = 0;
+    int     corners = 1;
 
     /*
      * Handle command line options.
@@ -50,6 +52,8 @@ int main(int argc, char *argv[])
     {
         if( strcmp(argv[i],"-no_norm") == 0 )
             norm_print_flag = 0;
+        else if( strcmp(argv[i],"-no_corners") == 0 )
+            corners = 0;
         else if( strcmp(argv[i],"-tfw") == 0 )
             tfw_flag = 1;
         else if( strcmp(argv[i],"-proj4") == 0 )
@@ -130,7 +134,8 @@ int main(int argc, char *argv[])
             
             TIFFGetField( tif, TIFFTAG_IMAGEWIDTH, &xsize );
             TIFFGetField( tif, TIFFTAG_IMAGELENGTH, &ysize );
-            GTIFPrintCorners( gtif, &defn, stdout, xsize, ysize, inv_flag, dec_flag );
+            if( corners )
+                GTIFPrintCorners( gtif, &defn, stdout, xsize, ysize, inv_flag, dec_flag );
         }
 
     }
-- 
2.39.1


From 468e2a178fa2c72ac7ae4a226abb76ac047e3774 Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sat, 26 Nov 2022 13:24:02 +0100
Subject: [PATCH 2/2] testlistgeo: do not print corner coordinates for data
 files with Cassini Soldner projection, whose formulas have changed in PROJ
 9.1.1 (fixes #81)

---
 libgeotiff/test/testlistgeo          |  4 ++--
 libgeotiff/test/testlistgeo_out.dist | 14 --------------
 2 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/libgeotiff/test/testlistgeo b/libgeotiff/test/testlistgeo
index 596301b..9a41e74 100755
--- a/libgeotiff/test/testlistgeo
+++ b/libgeotiff/test/testlistgeo
@@ -59,11 +59,11 @@ $EXE ${DATA_DIR}/ProjLinearUnitsGeoKey_9036.tif >>${OUT}
 echo "" >>${OUT}
 
 echo "Testing listgeo ProjectedCSTypeGeoKey_28191_cassini_soldner.tif" >> ${OUT}
-$EXE ${DATA_DIR}/ProjectedCSTypeGeoKey_28191_cassini_soldner.tif >>${OUT}
+$EXE -no_corners ${DATA_DIR}/ProjectedCSTypeGeoKey_28191_cassini_soldner.tif >>${OUT}
 echo "" >>${OUT}
 
 echo "Testing listgeo cassini_soldner.tif" >> ${OUT}
-$EXE ${DATA_DIR}/cassini_soldner.tif >>${OUT}
+$EXE -no_corners ${DATA_DIR}/cassini_soldner.tif >>${OUT}
 echo "" >>${OUT}
 
 echo "Testing listgeo ProjectedCSTypeGeoKey_27200_new_zealand_mapping_grid.tif" >> ${OUT}
diff --git a/libgeotiff/test/testlistgeo_out.dist b/libgeotiff/test/testlistgeo_out.dist
index 742f0fc..20221ce 100644
--- a/libgeotiff/test/testlistgeo_out.dist
+++ b/libgeotiff/test/testlistgeo_out.dist
@@ -299,13 +299,6 @@ Ellipsoid: 7010/Clarke 1880 (Benoit) (6378300.79,6356566.43)
 Prime Meridian: 8901/Greenwich (0.000000/  0d 0' 0.00"E)
 Projection Linear Units: 9001/metre (1.000000m)
 
-Corner Coordinates:
-Upper Left    (  440720.000, 3751320.000)  ( 40d47'28.08"E, 64d13'29.57"N)
-Lower Left    (  440720.000, 3751260.000)  ( 40d47'27.69"E, 64d13'27.64"N)
-Upper Right   (  440780.000, 3751320.000)  ( 40d47'32.51"E, 64d13'29.40"N)
-Lower Right   (  440780.000, 3751260.000)  ( 40d47'32.12"E, 64d13'27.47"N)
-Center        (  440750.000, 3751290.000)  ( 40d47'30.10"E, 64d13'28.52"N)
-
 Testing listgeo cassini_soldner.tif
 Geotiff_Information:
    Version: 1
@@ -348,13 +341,6 @@ Projection Method: CT_CassiniSoldner
    ProjFalseNorthingGeoKey: 126867.909000 m
 Projection Linear Units: 9001/metre (1.000000m)
 
-Corner Coordinates:
-Upper Left    (  440720.000, 3751320.000)  ( 40d47'28.08"E, 64d13'29.57"N)
-Lower Left    (  440720.000, 3751260.000)  ( 40d47'27.69"E, 64d13'27.64"N)
-Upper Right   (  440780.000, 3751320.000)  ( 40d47'32.51"E, 64d13'29.40"N)
-Lower Right   (  440780.000, 3751260.000)  ( 40d47'32.12"E, 64d13'27.47"N)
-Center        (  440750.000, 3751290.000)  ( 40d47'30.10"E, 64d13'28.52"N)
-
 Testing listgeo ProjectedCSTypeGeoKey_27200_new_zealand_mapping_grid.tif
 Geotiff_Information:
    Version: 1
-- 
2.39.1

